lib/include/cne_mutex_helper.h

```diff
@@ -30,19 +30,35 @@ extern "C" {
 static inline int
 cne_mutex_create(pthread_mutex_t *mutex, int flags)
 {
-    int ret = EFAULT;
-
-    if (mutex) {
-        pthread_mutexattr_t attr;
-
-        ret = pthread_mutexattr_init(&attr);
-        if (ret == 0) {
-            ret = pthread_mutexattr_settype(&attr, flags);
-            if (ret == 0) {
-                ret = pthread_mutex_init(mutex, &attr);
-                if (ret == 0 && pthread_mutexattr_destroy(&attr) == 0)
-                    return 0;
-            }
+    pthread_mutexattr_t attr;
+    int inited = 0, ret = EFAULT;
+
+    if (!mutex)
+        goto err;
+
+#define __do(_exp)    \
+    do {              \
+        ret = _exp;   \
+        if (ret)      \
+            goto err; \
+    } while (0 /* CONSTCOND */)
+
+    __do(pthread_mutexattr_init(&attr));
+    inited = 1;
+
+    __do(pthread_mutexattr_settype(&attr, flags));
+
+    __do(pthread_mutex_init(mutex, &attr));
+
+    __do(pthread_mutexattr_destroy(&attr));
+
+#undef __do
+
+    return 0;
+err:
+    if (inited) {
+        if (pthread_mutexattr_destroy(&attr)) {
```

Comment:
Why have an if statement here? I know you're trying to convince klocwork that you're checking the return value, but it seems like we should do something like log an error if this fails rather than check the return value just to ignore it.

Reply:
I can print out a message, but I did not want to lose the root failure.